### PR TITLE
fix(STONEINTG-835): don't update running gitlab commitStatus

### PR DESCRIPTION
**Pull request by dirgim**
Check if there is existing commitStatus and compare the status before calling r.client.Commits.SetCommitStatus. 
If the old and new status are both "Running", don't update commitStatus state

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: status/reporter_gitlab.go [hunk 1]
```diff
@@ -131,6 +131,20 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 		Description: gitlab.Ptr(report.Summary),
 	}
 
+	// Special case for gitLab `running` state because of a bug where it can't be updated to the same state again
+	if glState == gitlab.Running {
+		allCommitStatuses, _, err := r.client.Commits.GetCommitStatuses(r.projectID, r.sha, nil)
+		if err != nil {
+			return fmt.Errorf("error while getting all comments for merge-request %d: %w", r.mergeRequest, err)
+		}
+		existingCommitStatus := r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
```
Comment (MartinBasti) [inline status/reporter_gitlab.go:140]: don't we need a unittest here? it's the aim of this PR

**Review by MartinBasti — COMMENTED** [review]
code looks good

File: status/reporter_gitlab.go (hunk 1 above)
Comment (dirgim) [reply status/reporter_gitlab.go:140]: Good catch, added the unit test.

**Review by MartinBasti — APPROVED** [review]

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by jsztuka — APPROVED** [review]

**Pull request merged**